src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,48 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int INITIAL_SIZE = 0;
+    private static final int CAPACITY = 10;
+    private final K[] keys;
+    private final V[] values;
+    private int size;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[CAPACITY];
+        values = (V[]) new Object[CAPACITY];
+        size = INITIAL_SIZE;
+    }
+
     @Override
     public void put(K key, V value) {
+        boolean noEqualsElement = true;
+        for (int i = 0; i < size; i++) {
+            if ((key == null && keys[i] == null) || key != null && key.equals(keys[i])) {
```

Comment:
```suggestion
            if (key == keys[i] || key != null && key.equals(keys[i])) {
```
also this check could be moved to a separate method and reused in your program

Reply:
thank you, but I invented my "bicycle", but it was possible to make a method of comparison simply based on the image of OBJECTS